src/group_theory/free_group.lean

```diff
@@ -369,10 +369,19 @@ instance : has_mul (free_group α) :=
     (λ L₁ L₂ H, quot.induction_on y $ λ L₃, quot.sound $ red.step.append_right H)⟩
 @[simp] lemma mul_mk : mk L₁ * mk L₂ = mk (L₁ ++ L₂) := rfl
 
+def inv_rev (w : list (α × bool)) : list (α × bool) :=
+ (list.map (λ (g : α × bool), (g.1, bnot g.2)) w).reverse 
+
+lemma inv_rev_length : (inv_rev L₁).length = L₁.length :=
+begin
+  induction L₁, { finish, },
+  unfold inv_rev, { finish, },
+end
+
 instance : has_inv (free_group α) :=
-⟨λx, quot.lift_on x (λ L, mk (L.map $ λ x : α × bool, (x.1, bnot x.2)).reverse)
-  (assume a b h, quot.sound $ by cases h; simp)⟩
-@[simp] lemma inv_mk : (mk L)⁻¹ = mk (L.map $ λ x : α × bool, (x.1, bnot x.2)).reverse := rfl
+⟨λx, quot.lift_on x (λ L, mk (inv_rev L))
+  (assume a b h, quot.sound $ by { unfold inv_rev, cases h; simp })⟩
```

Comment:
You can use `quot.map` to get a slightly shorter (and more readable) definition.

Reply:
You are right!  d287acfd4257efb4a48e18d547ca1eb3331caec9 makes this change.